### PR TITLE
[3D] Allow instance picking for point clouds with history, fix other picking bugs

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
- Fixed clicking on individual points within LaserScan messages
- Fixed issues with clicking on PointCloud and LaserScan messages with Decay Time enabled
- Fixed LaserScan messages sometimes rendering incorrectly with Decay Time enabled

**Description**
- Previously we rendered **all** history entries when doing the 2nd picking pass, because the individually picked items were not Renderables (the only Renderable was the parent that held all history entries). Fixed this by making actual Renderable subclasses, `PointCloudRenderable` and `LaserScanRenderable`. (Fixes FG-4035)
- Instance picking for LaserScans was totally broken because the shader failed to compile. Fixed this by correcting the syntax. (Fixes FG-4244, probably conflicts with #6244)
- All LaserScan history entries for a topic were sharing materials, however, certain message data (start_angle/end_angle etc.) are set in the material uniforms, so new messages could affect the display of previously received scans if these values changed. Fixed this by creating per-renderable instances of the materials. (Fixes FG-4243)
- Fixed pixelRatio not being properly updated for picking materials

Relates to FG-3894
Relates to FG-3130
Relates to FG-4246

https://github.com/foxglove/studio/assets/14237/8a00cad9-4380-40e7-ad4a-cf7518e6d862

### CODE REVIEW

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/LaserScans.ts [hunk 1]
```diff
@@ -538,7 +575,7 @@ class LaserScanInstancePickingMaterial extends THREE.RawShaderMaterial {
         uniform float angleMin, angleIncrement;
         uniform float rangeMin, rangeMax;
         in float position; // range, but must be named position in order for three.js to render anything
-        varying vec4 objectId;
+        out vec4 objectId;
```
Comment (snosenzo) [inline packages/studio-base/src/panels/ThreeDeeRender/renderables/LaserScans.ts:578]: Is there a way that we can test that all of out custom shaders compile? or give a louder error when they don't?

Comment (jtbandes) [reply packages/studio-base/src/panels/ThreeDeeRender/renderables/LaserScans.ts:578]: At least testing the picking in a story makes sure the shader will execute…

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/PointClouds.ts [hunk 2]
```diff
@@ -959,21 +1011,34 @@ function createStixelGeometry(topic: string, usage: THREE.Usage): DynamicBufferG
...
+    );
+    // We don't calculate the bounding sphere for points, so frustum culling is disabled
+    stixels.frustumCulled = false;
+    stixels.name = `${userData.topic}:PointCloud:stixels`;
+    this.#stixels = stixels;
+    this.add(stixels);
+  }
+
+  public override dispose(): void {
+    this.#stixels.geometry.dispose();
```
[the remark this answers is not in the document]
Comment (jtbandes) [reply packages/studio-base/src/panels/ThreeDeeRender/renderables/PointClouds.ts:1038]: It seems there is just one stixelMaterial held by the PointCloudHistoryRenderable and disposed in PointCloudHistoryRenderable.dispose()

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/PointClouds.ts [hunk 3]
```diff
@@ -959,21 +1011,34 @@ function createStixelGeometry(topic: string, usage: THREE.Usage): DynamicBufferG
...
-  return stixels;
+class StixelsRenderable extends Renderable<BaseUserData, /*TRenderer=*/ undefined> {
+  #stixels: THREE.LineSegments<DynamicBufferGeometry, THREE.LineBasicMaterial>;
+  public readonly geometry: DynamicBufferGeometry;
+
+  public constructor(
+    name: string,
+    userData: BaseUserData,
+    geometry: DynamicBufferGeometry,
+    material: THREE.LineBasicMaterial,
```
[the remark this answers is not in the document]
Comment (jtbandes) [reply packages/studio-base/src/panels/ThreeDeeRender/renderables/PointClouds.ts:1022]: Currently all the StixelsRenderables share one material

**Pull request merged**